test/unit/dialects/better-sqlite3.js

```diff
@@ -33,10 +39,44 @@ describe('better-sqlite3 unit tests', () => {
         },
       });
 
-      const result = await knexInstance.select(knexInstance.raw('2 + 2 as answer')).first().catch(err => err);
+      const result = await knexInstance
+        .select(knexInstance.raw('2 + 2 as answer'))
+        .first()
+        .catch((err) => err);
       console.dir(result);
       expect(result).to.be.an('error');
       expect(result.code).to.equal('MODULE_NOT_FOUND');
     });
   });
+
+  describe('readonly', () => {
+    it('should initialize the DB with the passed-in `readonly` option', async () => {
+      const knexInstance = knex({
+        client: 'better-sqlite3',
+        useNullAsDefault: true,
+        connection: {
+          filename: '/Users/erindees/src/knex/test/test.sqlite3',
+          options: {
+            readonly: true,
+          },
+        },
+      });
+
+      const connection = await knexInstance.client.acquireConnection();
+      expect(connection.readonly).to.equal(true);
```

Comment:
can you add a clause that try to execute a query that modify data to check readonly option is take into account ?

Reply:
Yes, definitely. I originally wrote a test case for this, but deleted it because I thought it might be too "integration-y" for a unit test. But it is a good idea to check this, so I've restored it. Thank you for the feedback!